pkg/injection/codemodule/installer/job/installer.go

```diff
@@ -0,0 +1,121 @@
+package job
+
+import (
+	"context"
+	"os"
+
+	"github.com/Dynatrace/dynatrace-operator/pkg/controllers/csi/metadata"
+	"github.com/Dynatrace/dynatrace-operator/pkg/injection/codemodule/installer"
+	"github.com/Dynatrace/dynatrace-operator/pkg/injection/codemodule/installer/common"
+	"github.com/Dynatrace/dynatrace-operator/pkg/injection/codemodule/installer/symlink"
+	"github.com/Dynatrace/dynatrace-operator/pkg/util/kubeobjects/env"
+	jobutil "github.com/Dynatrace/dynatrace-operator/pkg/util/kubeobjects/job"
+	"github.com/pkg/errors"
+	"github.com/spf13/afero"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type Properties struct {
+	Owner        client.Object
+	ApiReader    client.Reader
+	Client       client.Client
+	ImageUri     string
+	PathResolver metadata.PathResolver
+	PullSecrets  []string
+}
+
+func NewInstaller(ctx context.Context, fs afero.Fs, props *Properties) installer.Installer {
+	return &Installer{
+		fs:       fs,
+		props:    props,
+		nodeName: env.GetNodeName(),
+	}
+}
+
+type Installer struct {
```

Comment:
why do you need to duplicate here installer.Installer? am I miss something?

Reply:
can we embed one type into another to add extra methods?

Reply:
the name of the package is `job`

when you import and use it:
```
"github.com/Dynatrace/dynatrace-operator/pkg/injection/codemodule/installer/job"

job.NewInstaller()
```

Reply:
> can we embed one type into another to add extra methods?

there is a common `interface` that is the `installer.Installer` 
- this is old code, didn't want to change it

Reply:
gotcha